File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.java
import java.util.*;

class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int len1 = word1.length();
        int len2 = word2.length();
        
        if (len1 > len2) {
            for (int i = 0; i < len2; i++) {
                sb.append(word1.charAt(i));
                sb.append(word2.charAt(i));
            }
            sb.append(word1.substring(len2));
        } else {
            for (int i = 0; i < len1; i++) {
                sb.append(word1.charAt(i));
                sb.append(word2.charAt(i));
            }
            sb.append(word2.substring(len1));
        }

        return sb.toString();        
    }
}

File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = ""
        if len(word1) <= len(word2):
            for i in range(len(word1)):
                ans += word1[i]
                ans += word2[i]
            ans += word2[i+1:]
        else:
            for i in range(len(word2)):
                ans += word1[i]
                ans += word2[i]
            ans += word1[i+1:]
            
        return ans